.changeset/curvy-otters-jog.md

```diff
@@ -0,0 +1,56 @@
+---
+'astro': minor
+---
+
+Refactor type for integration hooks to be extensible from an interface on the global `Astro` namespace.
+
+To declare your own hooks for integrations extend the interface by adding the following code in your integration:
```

Comment:
```suggestion
To declare your own hooks for your integration, extend the `Astro.IntegrationHooks` interface:
```
nit: we only say "add the following code" if it's something they should actually copy and paste. This is a generic model of the shape of code to provide, and "adding" it wouldn't help them.